src/herder/TxSetFrame.cpp

```diff
@@ -281,58 +281,82 @@ TxSetFrame::surgePricingFilter(Application& app)
 }
 
 bool
-TxSetFrame::checkOrTrim(
-    Application& app,
-    std::function<bool(TransactionFrameBasePtr, SequenceNumber)>
-        processInvalidTxLambda,
-    std::function<bool(std::deque<TransactionFrameBasePtr> const&)>
-        processInsufficientBalance)
+TxSetFrame::checkOrTrim(Application& app,
+                        std::vector<TransactionFrameBasePtr>& trimmed,
+                        bool justCheck)
 {
     LedgerTxn ltx(app.getLedgerTxnRoot());
 
+    std::unordered_map<AccountID, int64_t> accountFeeMap;
```

Comment:
can we use the same map for tracking total fees and tx queues? `AccountTransactionQueue` is already purely internal and only used in this context, so it seems right.

The typical case is that they're the same, so if we can avoid managing an extra unordered_map it's better.

Reply:
I was not saying to compute the fees all the time, just use the same map (the expensive bit is managing the `AccountID` keys).
`buildAccountTxQueues` would allocate entries with fee set to 0, but in the typical case, we would set all of those to something else here. For the few fee bump we may allocate empty deque, but this is minimal.

Reply:
this is fine. as my other comment (I think in CAP0028) we need to do some perf runs anyways before releasing v13 to make sure those small things don't add up to something big.